### DISCUSSION TITLE
STILT vs meteorology projection

### DISCUSSION THREAD

**Discussion by UCASY** [uataq/stilt#121] [Users]
Hello, professors：
I used the WRF output to driven the STILT model, I have 4 questions about it.
1. First question about the projection, the projection of the WRF output is lambert_conformal_conic, and I found the run_stilt.r projection <- '+proj=longlat', so, should the projection of the run_stilt.r must be consistent with the WRF output?
2. I found the spatial resolution in the run_stilt.r is degree, but in WRF output is m, the spatial resolution must consistent with the WRF output？
3. # Receptor location(s) 
lati <- 40.766189   
long <- -111.847672  
zagl <- 25       
the receptor location only can set one in once simulation？
How it can set multiple locations in once simulation? Just like
lati <- 40.766189  38.0 …..   
long <- -111.847672  -112.0 …
zagl <- 25 24 …
i got an error about it when set 2 locations
[1] "2021-04-01 12:00:00 UTC" "2021-04-01 13:00:00 UTC"
Error: unexpected numeric constant in "lati <- 43.88 44.0"
Execution halted
4. how to generate the emissions.rds file？

**Comment by DustinRoten** [comment]
Greetings!

(1) I normally use meteorology files from READY ARL which are in pre-formatted for STILT/HYSPLIT.  **Are you using a converter to ensure your WRF files are STILT compatible? **

(2) (see above)

(3) Typically, I use X-STILT which is a slightly different version of the standard STILT model. For multiple receptors, I have rewritten `run_stilt.r` as a function so I can loop through multiple receptors with `for(...)`.

(4) Are you wanting to generate the `emissions.rds` file separately? Or as part of the STILT output?

[@tartanrunner25 should be able to answer (1) and (2) better than I can.]

**Comment by UCASY** [comment]
Hello, Roten:
(1-2)I first run the wrf model and used the [arw2arl_batch.sh] in stilt-tutorials to convert the wrfout_2021* to wrfout*.arl.  and used the wrfout*.arl to driven the stilt model, and run: Rscript run_stilt.r, so the spatial resolution and projection of STILT output must consistent with the wrfout？
(3)Maybe the X-stilt more suitable for my research,  i will have a try.
(4) I want to generate the emissions.rds file before running the STILT model  and used it. just as the emissions.rds in https://github.com/uataq/stilt-tutorials/tree/main/01-wbb.
Thank you.

**Comment by tartanrunner25** [comment]
Hello,

For number 1 and 2, WRF is independent of STILT in that the WRF grid does not matter. STILT will generate trajectories by interpolating winds from the WRF Lambert conformal grid and using the interpolated winds to create the backward trajectories. The trajectory positions and their corresponding footprint values are then gridded to generate the gridded STILT footprint. So long story short, just because WRF is on a Lambert conformal grid does not mean STILT's footprint needs to be on the same grid. For my own WRF runs, I typically use the same grid projection as you, and then run STILT and project the footprints on a standard lat/lon grid. The grid can also be at a completely different resolution. 

For number 3, yes you can have multiple locations. When you create a vector of the runs you want in the run_stilt.r code, your data table of the runs you want to generate should look like the following (just an example):

run_time   lati     long zagl
1 2017-09-26 00:00:00 40.806 -104.756    5
2 2017-09-26 03:00:00 40.806 -104.756    5
3 2017-09-26 06:00:00 40.806 -104.756    5
4 2017-09-26 09:00:00 40.806 -104.756    5
5 2017-09-26 12:00:00 40.806 -104.756    5
6 2017-09-26 15:00:00 40.806 -104.756    5

If you want there to be multiple locations you would just need reconstruct the data table such that the second location has its own row and time, for example:

run_time   lati     long zagl
1 2017-09-26 00:00:00 40.806 -104.756    5         <- this is location 1
2 2017-09-26 03:00:00 45.806 -108.243    5.        <- this is location 2

As long as the run is in this data table format, the STILT wrapper is smart enough to set up a run for different locations. I've done this using 100+ receptors for one run_stilt.r job. 

For number 4, you will need to find your own emissions inventory and convolve is with the STILT footprint. The link that Dustin provided gives an example of how to convolve the emissions with the STILT footprint. For the US, we commonly use VULCAN for CO2 emissions. If you are looking for a global emission product, ODIAC might be a good place to start. I've also used STILT with fire emission inventories like FINN, really the options are pretty limitless =) 


-Derek

**Comment by UCASY** [comment]
I see. Thank you for your help!

**Comment by UCASY** [comment]
Hello, Professors：
  I have questions about the footprint. I read the code in the calc_footprint.r the definition is calc_footprint generates upstream influence footprint.  I want to use the forward function, but the the documentation didn't  show it can generates **downstream** influence footprint, although it can generate the foot*.nc files. So i confused in this. My question is any difference between backward footprint and forward footprint？is the model has the ability to generate downstream footprint？
the forward output foot*.nc can be called the footprint？Thank you.
Yang.

**Comment by tartanrunner25** [comment]
In essence, it can be view as forward footprint, so yes you can still use the foot*.nc files.

**Comment by UCASY** [comment]
Thank you.

**Comment by XIR-art** [comment]
Hello, UCASY user Mr. Yang.
On github, I learned that my research direction is similar to yours, using wrf-stilt to simulate CO2 concentration, can I have some questions to discuss with you? This is my email address:1101850860@qq.com
